File: Rohit Singhal/Trees/BinaryTreeLevelOrderTraversalII.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {

    public List<List<Integer>> levelOrderBottom(TreeNode root) {
        if (root == null) {
            List<List<Integer>> base = new ArrayList<>();
            return base;
        }
        List<List<Integer>> ans = new ArrayList<>();
        List<Integer> temp = new ArrayList<>();
        Queue<TreeNode> pd = new LinkedList<>();
        pd.add(root);
        pd.add(null);
        while (!pd.isEmpty()) {
            TreeNode top = pd.poll();

            if (top == null) {
                if (pd.isEmpty()) {
                    ans.add(temp);
                    break;
                } else {
                    ans.add(temp);
                    temp = new ArrayList<>();
                    pd.add(null);
                }
            } else {
                temp.add(top.val);
                if (top.left != null) pd.add(top.left);

                if (top.right != null) pd.add(top.right);
            }
        }
        Collections.reverse(ans);
        return ans;
    }
}

File: Aman/Tree/Binary Tree Level Order Traversal II.cpp
class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        
    vector<vector<int>>res;
       
    if(root==NULL)
        return res;
        
    queue<TreeNode*>q;
    vector<int>v;     
    q.push(root);
        
    while(!q.empty())
    {
       int n=q.size();
       for(int i=0;i<n;i++)
       {
           TreeNode* temp=q.front();
           q.pop();
           v.push_back(temp->val);
           if(temp->left)
             q.push(temp->left);
           if(temp->right)
             q.push(temp->right);
       }
       res.push_back(v);  
       v.resize(0);
    }
    reverse(res.begin(),res.end());
    return res;
    }
};
